packages/rocketchat-ui/client/lib/msgTyping.js

```diff
@@ -12,6 +12,13 @@ export const MsgTyping = (function() {
 	const selfTyping = new ReactiveVar(false);
 	const usersTyping = {};
 	const dep = new Tracker.Dependency;
+	let disableUserTyping = undefined;
+	Tracker.autorun(function() {
+		if (!Meteor.userId()) {
+			return;
+		}
+		disableUserTyping = RocketChat.getUserPreference(Meteor.userId(), 'disableUserTyping');
```

Comment:
If `Allow_DisableUserTyping` changes from `true` to `false`, a user can still have the `disableUserTyping` preference set to `true`.

```suggestion
		disableUserTyping = RocketChat.settings.get('Allow_DisableUserTyping') === true 
 ? RocketChat.getUserPreference(Meteor.userId(), 'disableUserTyping') : false;
```